toolkit/tools/imagegen/configuration/systemconfig.go

```diff
@@ -66,6 +67,34 @@ func (s *SystemConfig) GetMountpointPartitionSetting(mountPoint string) (partiti
 	return FindMountpointPartitionSetting(s.PartitionSettings, mountPoint)
 }
 
+func (s *SystemConfig) validateUsersAndGroups() (err error) {
+	for _, user := range s.Users {
+		groupMatchFunc := func(groupName interface{}, groupObj interface{}) bool {
+			return groupName == groupObj.(Group).Name
+		}
+
+		if user.PrimaryGroup != "" {
+			if !sliceutils.Contains(s.Groups, user.PrimaryGroup, groupMatchFunc) {
+				// Unclear how to validate these while allowing users to be added to existing system groups. If we
+				// define a group in the config that already exists it will cause errors.
+				// Maybe we can scrape /etc/group and /etc/passwd from filesystem.sepc to validate these?
```

Comment:
Can you be a bit more clear with the comment? What do you mean by "these" in "Unclear how to validate these" for instance? Also, I'm not sure, if the last comments is helpful. If we can check the files, let's do it. If not, why add the comment?

Reply:
```suggestion
				// Unclear how to validate that users and groups match together correctly while still allowing
				// users to be added to existing system groups. If we define a group in the config that already
				// exists it will cause errors, but we don't have insight into the default users/groups already in
				// an image before we install the filesystem package.
```